Controller/Router.php

```diff
@@ -50,9 +50,15 @@ public function match(\Magento\Framework\App\RequestInterface $request)
 
         $stylaFrontendName = $this->_getFrontendName();
         if (strpos($identifier, $stylaFrontendName) !== false) {
-
             $request->setModuleName('stylaconnect2page')->setControllerName('page')->setActionName('view');
-        } else {
+        } 
+        else if ($identifier == 'styla-plugin-version') {
+            $styla_version_arr = array();
+            $styla_version_arr['version'] = $this->_configHelper->getPluginVersion();
+            echo json_encode($styla_version_arr);
+            exit;
```

Comment:
The bailout using "exit" is quite crude, also there's no `application/json` content type header. What happens if an upstream method prints something that conflicts with the serialised json?